src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,206 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2018-2019 FIRST. All Rights Reserved.                        */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.subsystems;
+import java.util.logging.LogManager;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.LimitSwitchNormal;
+import com.ctre.phoenix.motorcontrol.LimitSwitchSource;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.ElevatorFeedforward;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DutyCycleEncoder;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.ElevatorConstants;
+import frc.robot.util.MotorFactory;
+import lib.ctre_shims.TalonEncoder;
+
+public class Elevator extends SubsystemBase {
+
+  private final WPI_TalonFX m_motor;
+  private final PIDController m_elevatorPID;
+  private final ElevatorFeedforward m_elevatorFF;
+  private final TalonEncoder m_talonEncoder; 
+  private final DigitalInput m_topLimitSwitch; 
+  private final DigitalInput m_bottomLimitSwitch; 
+  private final ShuffleboardTab m_elevatorTab; 
+  private double m_maxHeight; 
+  private boolean m_enabled; 
+  private boolean m_isCalibrated = false; 
```

Comment:
I think the idea of waiting for the robot to be calibrated before the PID runs is not a good idea. also max height isn't used for anything besides printing the max height.

Reply:
yeah I have used max height to make sure that any setpoints I place in are not over the elevator's max extension. Also max height is really the elevator's extension in meters.

Reply:
I want to calibrate before to make sure that the encoders are zeroed and so nothing funny happens when the PID is enabled in moveHeight().